addon/globalPlugins/webAccess/gui/ruleEditor.py

```diff
@@ -661,38 +768,368 @@ def initPropertiesList(self, context):
 		self.onInitUpdateListCtrl()
 		self.focusListCtrl(index=index)
 
-
 	def updateListCtrl(self, data):
 		super(PropertiesPanel, self).updateListCtrl(data)
 
-
-	def updateData(self, data = None):
+	def updateData(self, data=None):
 		propertiesMapValue = {}
-		data = self.context["data"]["rule"]
+		data = self.getRule()
 		ruleType = data.get("type")
 		if ruleType is not None and self.propertiesList:
 			for props in self.propertiesList:
 				propertiesMapValue[props.get_id()] = props.get_value()
-			if data.get("properties"):
-				del data["properties"]
```

Comment:
No need to delete here. Setting will overwrite any existing value.
Actually, the whole operation could be accomplished using _dict comprehension_ without the need for the intermediate `propertiesMapValue` variable.
Furthermore, we'd probably better only store properties that have a non-default value to increase JSON human-readability.